quic/s2n-quic/src/tests.rs

```diff
@@ -691,3 +699,105 @@ fn increasing_pto_count_under_loss() {
         pto_count
     );
 }
+
+// TODO: https://github.com/aws/s2n-quic/issues/1726
+//
+// The rustls tls provider is used on windows and has different
+// build options than s2n-tls. We should build the rustls provider with
+// mTLS enabled and remove the `cfg(target_os("windows"))`.
+#[cfg(not(target_os = "windows"))]
+#[test]
+fn mtls() {
+    use crate::provider::{event::events::HandshakeStatus, tls::default as s2n_tls};
+
+    // Ensure connection is successful under different network conditions
+    for delay_time in (500..=10_000).step_by(1000) {
+        for drop_percent in (0..=100).step_by(10) {
+            let delay_time = Duration::from_micros(delay_time);
+
+            let model = Model::default();
+            model.set_delay(delay_time);
+            // Only the Server sends a HandshakeDone frame
+            let server_subscriber = HandshakeStatusRecorder::new();
+            let frame_sent = server_subscriber.events();
+            // Only the Client receives a HandshakeDone frame
+            let client_subscriber = HandshakeStatusRecorder::new();
+            let frame_received = client_subscriber.events();
+
+            test(model.clone(), |handle| {
+                spawn(async move {
+                    // allow for 1 RTT worth of data before impairing the connection.
```

Comment:
This comment isn't correct in what's happening

Reply:
I meant to use `delay` not model.set_delay :( Does it make sense to rename `delay` to `sleep`?